Merge request from Pengfei about C4 functions

pengfeili1 opened the pull request:
Hi Jason,

I have made the following edits about ParmEd:

1. Add the onlywater option for add12_6_4 command;
2. add the changeC4AtomTypePair command, which changes the atom-type-pair C4 values;
3. fix a typo in the changeLJPair command.

Can you approve my pull request? Thanks!

Best,
Pengfei

pengfeili1 committed:
Add the support for four new water models OPC3, OPC, TIP3P-FB, and TIP4P-FB to add12_6_4 function in ParmEd.

pengfeili1 committed:
Update the test files to be consistent with the updated add12_6_4 function

pengfeili1 committed:
Merge branch 'ParmEd:master' into master

pengfeili1 committed:
1. Add the onlywater option for add12_6_4 command; 2. add the changeC4AtomTypePair command, which adds the atom-type-pair C4 values; 3. fix a typo in the changeLJPair command.

swails commented:
Can you merge the latest master branch and add a test case to ensure we don't break this feature?

pengfeili1 committed:
Add a test case for each of (1) onlywater option in add12_6_4 and (2) ChangeC4AtomTypePair command.

swails commented:
A few of the tests need to be updated (the `repr()` check can just be removed - it's unimportant, and the relative error check can be loosened so it passes).

I can do this if you can't, but won't have time until end of March, probably.

pengfeili1 committed:
remove the the repr() check and loosen the relative error check for the test_add12_6_4_onlywater and test_change_c4_atom_type_pair.

pengfeili1 commented:
Hi Jason,

I have removed the the repr() check and loosen the relative error check for the test_add12_6_4_onlywater and test_change_c4_atom_type_pair cases.

Let me know whether it works or not.

Best,
Pengfei

> On Mar 12, 2026, at 9:59 AM, Jason Swails ***@***.***> wrote:
> 
> 
> swails
>  left a comment 
> (ParmEd/ParmEd#1421)
>  <https://github.com/ParmEd/ParmEd/pull/1421#issuecomment-4047440461>
> A few of the tests need to be updated (the repr() check can just be removed - it's unimportant, and the relative error check can be loosened so it passes).
> 
> I can do this if you can't, but won't have time until end of March, probably.
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/ParmEd/ParmEd/pull/1421#issuecomment-4047440461>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ABUMM2KCAXNBMY3NCYMVQID4QLGGRAVCNFSM6AAAAACPTFTGKOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHM2DANBXGQ2DANBWGE>.
> You are receiving this because you authored the thread.
>

swails commented:
I meant the ones that are failing (see [here](https://github.com/ParmEd/ParmEd/actions/runs/23030970343/job/66889063810))

You should add back the `str()` commands to at least make sure the `__str__` method doesn't throw an exception.

pengfeili1 committed:
Add the str() back for the two test cases.

pengfeili1 commented:
When I click the link, it just tells me that there is error, but I could not find the details.

I thought you meant deleting the str() commands for the repr() check issue, because I did not find any repr() in the two test cases. I just added the str() back. If you meant other code except the two cases I added, I could not do that because I am afraid to mess up other code.

I will leave it for you to do the fixes: by the end of March is fine, as long as the new commands appear in the AmberTools2026 release.

> On Mar 13, 2026, at 11:12 AM, Jason Swails ***@***.***> wrote:
> 
> 
> swails
>  left a comment 
> (ParmEd/ParmEd#1421)
>  <https://github.com/ParmEd/ParmEd/pull/1421#issuecomment-4056271632>
> I meant the ones that are failing (see here <https://github.com/ParmEd/ParmEd/actions/runs/23030970343/job/66889063810>)
> 
> You should add back the str() commands to at least make sure the __str__ method doesn't throw an exception.
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/ParmEd/ParmEd/pull/1421#issuecomment-4056271632>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ABUMM2J74QK2AVUCB4H5ECL4QQXNXAVCNFSM6AAAAACPTFTGKOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHM2DANJWGI3TCNRTGI>.
> You are receiving this because you authored the thread.
>

swails commented:
#1430 replaces this PR (incorporates its commits).
